sklearn/tests/test_metaestimators_metadata_routing.py

```diff
@@ -35,7 +40,7 @@
 rng = np.random.RandomState(42)
 N, M = 100, 4
 X = rng.rand(N, M)
-y = rng.randint(0, 2, size=N)
+y = rng.randint(0, 3, size=N)
```

Comment:
All estimators support multiclass: https://scikit-learn.org/stable/modules/multiclass.html#multiclass-classification

This said, we also have a `binary_only` tag in the estimator.

Reply:
@adrinjalali @glemaitre Please, could you tell me whether I should work on passing two different `y`'s? 

I have tried to find out about estimator tags and I couldn't find `binary_only` implemented anywhere except for in the tests.

Reply:
In theory, you would have something like:

```python
from sklearn.utils._tags import _safe_tags

tags = _safe_tags(estimator)
if tags["binary_onlty"]:
    estimator.fit(X, y_binary)
else:
    estimator.fit(X, y_multiclass)
```

where `y_binary` and `y_multiclass` would be the created target at the beginning of the file.

So I foresee 2 solutions.

Create a `ConsumerBinaryClassifier` such that:

```python
class ConsumerBinaryClassifier(ConsumerClassifier):
    def fit(self, X, y, **fit_params):
        y_type = type_of_target(y)
        if y_type != "binary":
            raise ValueError("This classifier supports only binary target")
        return super().fit(X, y, fit_params)
    def _more_tags(self):
        return {"binary_only": True}
```

Otherwise, we go from the principle that there are no binary classification and we just add a `# TODO` comment to modify the code if one day it happens.

@adrinjalali what do you think is best.